### PR TITLE
chore: add a development explainer

### PR DESCRIPTION
**Pull request by thedadams**
This change also adds a docker build step to CI and adds the http/ directory to the gitignore.

Issue: https://github.com/obot-platform/obot/issues/909

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+Obot has a set of packaged tools. These tools are in the repo `github.com/obot-platform/tools`. By default, Obot will pull the tools from this repo. However, when developing tools in this repo, you can follow these steps to use a local copy.
+
+1. Clone `github.com/obot-platform/tools` to your local machine.
+2. In the root directory of the tools repo on your local machine, run `make build`.
+3. Run the Obot server, either with `make dev` or in your IDE, with the `OBOT_SERVER_TOOL_REGISTRY` environment variable set to the root directory of the tools repo.
+
+Now, any time one of these tools is run, your local copy will be used.
+
+> [!IMPORTANT]
+> Any time you change a Go based tool in your local repo, you must run `make build` in the tools repo for the changes to take effect with Obot.
```
Comment (StrongMonkey) [inline DEVELOPMENT.md:58]: Well I usually only build that binary specifically for the tool I care instead of building all of it.

**Review by StrongMonkey — APPROVED** [review]

**Review by ryanhopperlowe — APPROVED** [review]
🙌

**Review by drpebcak — APPROVED** [review]

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+The easiest way to run Obot locally is to run `make dev`. This will launch three processes: the API server, admin UI, and user UI. Opening `http://localhost:8080/admin/` will launch the admin UI. Changing the UI code will update the UI automatically. Changing any of the Go code requires restarting the server.
+
+## Building and Running the Obot Docker Image
+
+Obot is ultimately packaged into an image for distribution. You can build said image with `docker build -t my-obot .`, and then run the image via `docker run -p 8080:8080 my-obot`.
+
+## Debugging Obot
+
+It is possible to run the server and/or UIs in and IDE for debugging purposes. These steps layout what is necessary for Jet Brains IDEs, but an equivalent process can be used with VSCode based editors.
```
Comment (g-linville) [inline DEVELOPMENT.md]: tiny nits

```suggestion
It is possible to run the server and/or UIs in and IDE for debugging purposes. These steps layout what is necessary for JetBrains IDEs, but an equivalent process can be used with VSCode-based editors.
```

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+To run the server in GoLand:
+1. Create a new "Go Build" configuration.
+2. In the "Program Arguments" section, enter `server --dev-mode`.
+
+Then you're ready to run or debug this target.
+
+### Admin UI
+
+To run the Admin UI in GoLand or Web Storm:
```
Comment (g-linville) [inline DEVELOPMENT.md]: tiny nit:

```suggestion
To run the Admin UI in GoLand or WebStorm:
```

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+2. In the "package.json" dropdown, select the `package.json` file in the `ui/admin` directory.
+3. In the "Command" dropdown, select `run`.
+4. In the "Scripts" dropdown, select `dev`.
+5. In the "Environment" section, enter `VITE_API_IN_BROWSER=true`.
+
+Then you're ready to run or debug this target.
+
+### User UI
+
+To run the User UI in GoLand or Web Storm:
```
Comment (g-linville) [inline DEVELOPMENT.md]: tiny nit:

```suggestion
To run the User UI in GoLand or WebStorm:
```

File: DEVELOPMENT.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+## Obot Server Dev Mode
+
+In the description above for running the server in an IDE, the `--dev-mode` flag is used. This flag is also used when running the server with `make dev`. This does a few things (like turns on debug logging), the most helpful of which is to give you access to the database via `kubectl`. The kubeconfig is located at `tools/devmode-kubeconfig`.
+
+For example, from the root directory of the obot repo, you can list all agents in your setup with `kubectl --kubeconfig tools/devmode-kubeconfig get agents`.
+
+## Obot Credentials
+
+The GPTScript credentials for Obot are, by default, store in an SQLite database called `obot-credentials.db` in the root of the obot repo. You can use the `sqlite3` CLI to inspect the database directly: `sqlite3 obot-credentials.db`.
```
Comment (g-linville) [inline DEVELOPMENT.md]:
```suggestion
The GPTScript credentials for Obot are, by default, stored in a SQLite database called `obot-credentials.db` in the root of the obot repo. You can use the `sqlite3` CLI to inspect the database directly: `sqlite3 obot-credentials.db`.
```

I know it's probably technically pronounced "S Q lite", but I'm pretty sure most people just say "sequelite".

**Review by g-linville — APPROVED** [review]
just some docs nits

File: .github/workflows/docker-build.yml [hunk 6]
```diff
@@ -1,75 +1,26 @@
...
-      uses: joelwmale/webhook-action@2.4.1
-      with:
-        url: ${{ secrets.RENDER_MAIN_DEPLOY_URL }}
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Depot
+        uses: depot/setup-action@v1
+
+      - name: Build and push Docker image
```
Comment (thedadams) [inline .github/workflows/docker-build.yml]: Note for myself to fix this
```suggestion
      - name: Build Docker Image
```

File: DEVELOPMENT.md (hunk 2 above)
Comment (thedadams) [reply DEVELOPMENT.md]: Done.

File: DEVELOPMENT.md (hunk 3 above)
Comment (thedadams) [reply DEVELOPMENT.md]: Done.

File: DEVELOPMENT.md (hunk 4 above)
Comment (thedadams) [reply DEVELOPMENT.md]: Done.

File: DEVELOPMENT.md (hunk 5 above)
Comment (thedadams) [reply DEVELOPMENT.md]: Done.

File: .github/workflows/docker-build.yml (hunk 6 above)
Comment (thedadams) [reply .github/workflows/docker-build.yml]: Done.

**Pull request merged**